pkg/scheduler/plugins/proportion/proportion.go

```diff
@@ -239,7 +239,13 @@ func (pp *proportionPlugin) OnSessionOpen(ssn *framework.Session) {
 			metrics.UpdateQueueDeserved(attr.name, attr.deserved.MilliCPU, attr.deserved.Memory)
 		}
 
-		remaining.Sub(increasedDeserved).Add(decreasedDeserved)
+		remaining.Add(decreasedDeserved)
+		if increasedDeserved.LessEqual(remaining, api.Zero) {
+			remaining.Sub(increasedDeserved)
+		} else {
+			klog.V(4).Infof("Exiting when increasedDeserved <%v> is greater than remaining <%v>", increasedDeserved, remaining)
```

Comment:
Suggest to reset some kind of resouce within`remaining` to be `0` is the result is a negative value . Because it has added `decreasedDeserved` above left. When jump out of the judgement, it will judge the following condition `remaining.IsEmpty()` below, which may cause wrong result.